README.md

```diff
@@ -40,6 +40,138 @@ Then require the CSS in your `application.css` file:
+</form>
+```
+
+#### Important Differences Between `form_with` and `form_for`
+
+Rails 5.1 introduced `form_with`,
+which unifies the functionality previously found in `form_for` and `form_tag`.
+`form_for` and `form_tag` will be deprecated in a future version of Rails,
+so new applications should use `bootstrap_form_with`.
+
+`form_with` is different compared to `form_for` and `form_tag`.
+`bootstrap_form_width` basically just wraps `form_with`
+and adds some functionality,
+and so the different behaviour of `form_with`
+is reflected in `bootstrap_form_with`
+compared to `bootstrap_form_for`
+and `bootstrap_form_tag`.
+
+##### Ajax by Default
+`form_with` defaults to submitting forms via Javascript XHR calls,
+like `form_for` or `form_tag` would do if you specified `remote: true`.
+If you want the browser to submit the request
+the same way `form_for` and `form_tag` would do by default,
+you need to specify `local: true` as an option to `form_with`.
+
+##### No Default DOM IDs
+When used with the builder (variable) yielded by `form_with`,
+the Rails field helpers do not generate a default DOM id.
+Because `bootstrap_form_width` just wraps and adds some functionality
+to `form_with`,
+the `bootstrap_form_with` field helpers also do not generate a default DOM id.
+This affect how labels work in your application,
+and may affect automated testing if you're using Capybara or similar tools,
+and you wrote actions or tests that selected on the DOM id of an element.
+
+For Rails 5.1, you can specify the id explicitly in most cases:
+
+```erb
+<%= bootstrap_form_with(model: @user, local: true) do |f| %>
+  <%= f.email_field :email %>
+  <%= f.password_field :password, id: :password %>
+  <%= f.submit "Log In" %>
+<% end %>
+```
+
+generates:
+
+```html
+<form role="form" action="/users" accept-charset="UTF-8" method="post">
+  <input name="utf8" type="hidden" value="&#x2713;" />
+  <div class="form-group">
+    <label class="required" for="user_email">Email</label>
+    <input class="form-control" type="email" value="steve@example.com" name="user[email]" />
+  </div>
+  <div class="form-group">
+    <label for="password">Password</label>
+    <input id="password" class="form-control" type="password" name="user[password]" />
+    <small class="form-text text-muted">A good password should be at least six characters long</small>
+  </div>
+  <input type="submit" name="commit" value="Log In" class="btn btn-secondary" data-disable-with="Log In" />
+</form>
+```
+
+##### Use `fields` Instead Of `fields_for` In Nested Forms
+For nested forms, use `fields` instead of `fields_for`.
+
+##### No Default Classes
+Finally, `bootstrap_form_with` doesn't attach a default class
+to the form.
+If you attached styling to the DOM class that `form_for` added to the form element,
+you'll have to add your own code to attach the appropriate class.
+
+#### Nested Forms with `bootstrap_form_with`
+
+This hasn't been tested yet.
+
+### Rails < 5.1
+
```

Comment:
Some of this is documentation of `form_with`. Readme should contain info relevant to the library, and not what it wraps. Nobody wants to read a wall of text. Duplicating rendered html produced by bootstrap_form_for and bootstrap_form_with is redundant.

Reply:
I agree that there's a lot of HTML in the README. I was just following the outline of the documentation for `form_for`.